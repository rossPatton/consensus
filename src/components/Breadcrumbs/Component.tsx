import React, { memo } from 'react';
import { Link } from 'react-router-dom';

export const BreadcrumbsComponent = memo((props: any) => (
  <ul className="fs6 fw600 mB2 lh1 lsNone fx fxWrap brdB1 mB3 pB3">
    {console.log('props => ', props)}
    {props.country && (
      <li className="mR1">
        {!props.region && props.country.name}
        {props.region && (
          <Link to={`/${props.country.code}`}>
            {props.country.name}
          </Link>
        )}
      </li>
    )}
    {props.region && (
      <li className="mHide mR1">/</li>
    )}
    {props.region && (
      <li className="mR1">
        {!props.city && props.country.name}
        {props.city && (
          <Link to={`/${props.country.code}/${props.region.code}`}>
            {props.region.name}
          </Link>
        )}
      </li>
    )}
    <li className="mHide mR1">/</li>
    <li className="mR1">
      <Link to="/us/ny/new-york-city">
        New York City
      </Link>
    </li>
  </ul>
));
