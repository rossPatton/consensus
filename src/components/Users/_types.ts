import {match} from 'react-router';

type tMatch = match & {
  params: {
    page?: string,
    [key: string]: string,
  }
};

export type tState = {
  role: tRole,
};

export type tRoleOpts = {
  orgId: number,
  role: tRole,
  userId: number,
};

type tProps = {
  removeUser?: (ev: React.MouseEvent<HTMLButtonElement>, id: number) => void,
  setUserRole?: (ev: React.ChangeEvent<HTMLSelectElement>, id: number) => void,
  users: tUser[],
  sessionRole: tRole, // to distinguish from user roles, and to shut up eslint
}

export interface tContainerProps extends tProps {
  match: tMatch,
}

export interface tComponentProps extends tProps {
  isEditable: boolean,
}
