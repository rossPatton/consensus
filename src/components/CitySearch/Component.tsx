import _ from 'lodash';
import React, {memo} from 'react';

import stateCodeMap from '../../json/usa/stateCodeMap.json';
import {tComponentProps} from './_types';

export const CitySearchComponent = memo((props: tComponentProps) => {
  const onCityChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const city = _.find(props.cities, c => c.name === ev.currentTarget.value);
    if (city) {
      props.updateState({
        city: city.name,
        cityId: city.id,
        region: city.region,
        regionId: city.regionId,
      });
    }
  };

  const onRegionChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const region = ev.currentTarget.value;
    if (region !== props.region) {
      props.updateState({
        city: '',
        cityId: 0,
        region,
        regionId: 0,
      });
    }
  };

  return (
    <>
      {props.showRegionField && (
        <>
          <h2 className="fs5 mB1 lh1">
            Pick a different state
          </h2>
          <label htmlFor="stateSelect">
            <select
              name="type"
              id="stateSelect"
              className="mB4 row bgWhite"
              value={props.region}
              onBlur={onRegionChange}
              onChange={onRegionChange}>
              {Object.keys(stateCodeMap).map(state => (
                <option key={state} value={state}>
                  {state}
                </option>
              ))}
            </select>
          </label>
        </>
      )}
      <h2 className="fs5 mB1 lh1">
        City in <span className="dInBl mR2">
          {(!props.showRegionField && props.geo.region) && props.geo.region}
          {(props.showRegionField && props.region) && props.region}
        </span>
        {!props.showRegionField && (
          <button
            type="button"
            onClick={props.toggleRegionField}>
            Not the right state?
          </button>
        )}
      </h2>
      {props.label && (
        <p className="fs5 copyBlack mB1">
          {props.label}
        </p>
      )}
      <div className="mB4">
        {props.cityId === 0 && (
          <label htmlFor="citySelect">
            <select
              name="type"
              id="citySelect"
              className="row bgWhite"
              value={props.city}
              onBlur={onCityChange}
              onChange={onCityChange}>
              {props.cities.map((city: tCity) => (
                <option key={city.id} value={city.name}>
                  {city.name}, {city.region}
                </option>
              ))}
            </select>
          </label>
        )}
        {props.cityId > 0 && (
          <div className="brdA1 p3 black br8 dInBl">
            <b>{props.city}</b>, <span className="dInBl mR3">{props.region}</span>
            <button
              type="button"
              onClick={() => {
                props.updateState({
                  city: '',
                  cityId: 0,
                });
              }}>
              Not the right city?
            </button>
          </div>
        )}
      </div>
    </>
  );
});
