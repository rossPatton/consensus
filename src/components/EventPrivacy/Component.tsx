import React, {memo} from 'react';

import {tProps} from './_types';

export const EventPrivacyComponent = memo((props: tProps) => (
  <small className="bgYellowLite br8 p1 pL2 pR2 mR2">
    {props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Lock Emoji">
          ðŸ”’
        </span>
        Private Event
      </>
    )}
    {!props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Tada Emoji">
          ðŸŽ‰
        </span>
        Public Event
      </>
    )}
  </small>
));
