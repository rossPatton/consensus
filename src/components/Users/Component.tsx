import _ from 'lodash';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {tComponentProps} from './_types';
const roles: tRole[] = ['member', 'facilitator'];

export const UsersComponent = memo((props: tComponentProps) => (
  <ul>
    {props.users.map((user: tUser, i) => (
      <li
        key={i}
        className="bg-white mb-3 hover:bg-gray-11 br4 flex items-center p-3">
        <div className="bgGrey3 circ mr-3 p-3" />
        <div className="w-full">
          <h3 className="mb-1">
            <Link to={`/user/${user.id}`}>
              @{user.username}
            </Link>
          </h3>
          <div className="flex flex-col d:flex-row items-center">
            {user.role !== 'pending' && (
              <span className="p-1 leading-none fs7 bgGrey4 white text-bold br4 mr-2">
                {user.role === 'member' ? props.memberName : props.modName}
              </span>
            )}
            {(props.sessionRole === 'admin' || props.sessionRole === 'facilitator')
                && (
                  <button
                    className="fs7 bg-white"
                    onClick={ev => props.removeUser(ev, user.id)}>
                    {user.role === 'pending'
                      ? 'Reject this user'
                      : 'Remove this user'}
                  </button>
                )}
          </div>
        </div>
        {(props.sessionRole === 'admin' || props.sessionRole === 'facilitator')
          && (
            <select
              className="capitalize"
              value={user.role as string}
              onChange={ev => props.setUserRole(ev, user.id)}>
              <option key="" value={user.role}>
                {user.role === 'pending'
                  ? 'Approve'
                  : 'Change role'}
              </option>
              {roles.map(role => (
                role === user.role
                  ? null
                  : (
                    <option
                      key={role}
                      value={role}>
                      {role === 'member' ? props.memberName : props.modName}
                    </option>
                  )
              ))}
            </select>
          )}
      </li>
    ))}
  </ul>
));
