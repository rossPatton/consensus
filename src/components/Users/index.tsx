import _ from 'lodash';
import React from 'react';
import {connect} from 'react-redux';

import {GenericLoader, Paginate, RoleFilter, SearchFilter} from '~app/containers';
import {
  checkSuccess,
  deleteUserByGroupId,
  getUsersByGroupId,
  patchUserByGroupId,
} from '~app/redux';

import {tContainerProps, tState, tStore} from './_types';
import {UsersComponent} from './Component';

class UsersContainer extends React.PureComponent<tContainerProps, tState> {
  static defaultProps = {
    className: '',
  };

  state = {
    allSelected: false,
    showMobileControls: null as number | null,
  };

  constructor (props: tContainerProps) {
    super(props);
    const groupId = props.group.id;
    props.getUsersByGroupIdDispatch({groupId, noPending: 'false'});
  }

  toggleAll = (users: ts.user[]) => {
    const selected = !this.state.allSelected;
    let checked = {...this.props.checked};
    if (selected) {
      for (const user of users) {
        checked[user.id] = selected;
      }
    } else {
      checked = {};
    }

    this.props.dispatch(checkSuccess(checked));
    this.setState({
      allSelected: selected,
    });
  }

  toggleCheck = (userId: number) => {
    const isChecked = !this.props.checked[userId];

    let checked = {...this.props.checked};
    if (isChecked) {
      checked = {
        ...this.props.checked,
        [userId]: !this.props.checked[userId],
      };
    } else {
      delete checked[userId];
    }

    this.props.dispatch(checkSuccess(checked));
  }

  removeUser = (ev: React.MouseEvent<HTMLButtonElement>, userId: number) => {
    ev.preventDefault();
    const {deleteUserByGroupIdDispatch, group} = this.props;
    deleteUserByGroupIdDispatch({
      userId,
      groupId: group.id,
    });
  }

  setUserRole = (ev: React.ChangeEvent<HTMLSelectElement>, userId: number) => {
    ev.preventDefault();
    const role = ev.currentTarget.value as ts.role;
    const groupId = this.props.group.id;
    if (groupId && userId && role) {
      this.props.patchUserByGroupIdDispatch({role, groupId, userId});
    }

    this.setState({
      showMobileControls: null,
    });
  }

  toggleMobileControls = (index: number) => {
    let showMobileControls = index;
    if (index === this.state.showMobileControls) {
      showMobileControls = null;
    }

    this.setState({
      showMobileControls,
    });
  }

  render() {
    const {
      count = 10,
      sessionRole,
      type,
      usersByGroupIdThunk,
    } = this.props;

    const isAdmin = sessionRole === 'admin' || sessionRole === 'facilitator';
    const isEditable = isAdmin && this.props.isEditable;
    const isSelectable = isAdmin && this.props.isSelectable;

    const {data: allUsers, isLoading} = usersByGroupIdThunk;
    let itemsToRender = allUsers.filter(u => u.role !== 'pending');
    if (type === 'pending') {
      const approvals: ts.user[] = allUsers.filter(u => u.role === 'pending');
      itemsToRender = approvals;
    }

    if (itemsToRender.length === 0) {
      return (
        <>
          {type === 'pending' && 'No users pending approval.'}
          {type !== 'pending' && 'No users found that meet this criteria.'}
        </>
      );
    }

    return (
      <GenericLoader
        isLoading={isLoading}
        render={() => (
          <RoleFilter
            items={itemsToRender}
            render={roleProps => (
              <SearchFilter
                searchKey="username"
                items={roleProps.items}
                render={searchProps => (
                  <Paginate
                    count={count}
                    items={searchProps.items}
                    render={(usersToRender: ts.user[]) => (
                      <UsersComponent
                        // passed in props
                        {...this.props}
                        {...this.state}
                        isEditable={isEditable}
                        isSelectable={isSelectable}
                        onRoleFilterChange={roleProps.onRoleFilterChange}
                        onSearchChange={searchProps.onSearchChange}
                        removeUser={this.removeUser}
                        setUserRole={this.setUserRole}
                        toggleAll={this.toggleAll}
                        toggleCheck={this.toggleCheck}
                        toggleMobileControls={this.toggleMobileControls}
                        users={usersToRender}
                      />
                    )}
                  />
                )}
              />
            )}
          />
        )}
      />
    );
  }
}

const mapStateToProps = (store: tStore) => ({
  checked: store.checked,
  usersByGroupIdThunk: store.usersByGroupId,
});

const mapDispatchToProps = (dispatch: Function) => ({
  dispatch,
  getUsersByGroupIdDispatch: (query: ts.usersByGroupIdQuery) =>
    dispatch(getUsersByGroupId(query)),

  deleteUserByGroupIdDispatch: (query: ts.deleteUserByGroupIdQuery) =>
    dispatch(deleteUserByGroupId(query)),

  patchUserByGroupIdDispatch: (query: ts.patchUserRoleQuery) =>
    dispatch(patchUserByGroupId(query)),
});

const Users = connect(mapStateToProps, mapDispatchToProps)(UsersContainer);
export default Users;
