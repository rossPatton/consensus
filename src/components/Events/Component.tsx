import cx from 'classnames';
import dayJS from 'dayjs';
import _ from 'lodash';
import qs from 'query-string';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {ExternalLink, PlaceholderImage, RSVP} from '../../components';
import {objToQueryString} from '../../utils';
import {tComponentProps} from './_types';

export const EventsComponent = memo((props: tComponentProps) => (
  <ul
    className={cx({
      'fx fxdRow taL': props.horizontal,
      'jcBetween': props.horizontal && props.events.length === 4,
    })}>
    {props.events.map((ev, i) => {
      const isPastEvent = dayJS(ev.date).isBefore(dayJS());

      return (
        <li
          key={ev.id}
          className={cx({
            'mB2': !props.horizontal,
            'col row': props.horizontal,
            'mR4': props.horizontal && i !== props.events.length - 1,
          })}>
          <div
            className={cx({
              'fx aiCtr': !props.horizontal,
              'p3 hvrBgGrey1 br8': props.isEditable,
            })}>
            <div
              className={cx({
                'mB2 bgWhite': props.horizontal,
                'mR3 bgGrey2': !props.horizontal,
              })}>
              <Link
                className="noUnderline"
                to={ev.isDraft
                  ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                  : `/event/${ev.id}`}>
                <PlaceholderImage
                  height={100}
                  seed={ev.id}
                  width={200}
                />
              </Link>
            </div>
            <div>
              <div
                className={cx({
                  'fx aiCtr mB1 fs7 fw600 lh1': true,
                  o5: isPastEvent,
                })}>
                <time className="mR1" dateTime={ev.date}>
                  {dayJS(ev.date).format('MMM DD | h:mmA')}
                </time>
                {!props.horizontal && (
                  <>
                    <span className="mR1">@</span>
                    {ev.locationLink && (
                      <ExternalLink
                        noFollow
                        className="mR1"
                        to={ev.locationLink}>
                        {ev.location}
                      </ExternalLink>
                    )}
                  </>
                )}
                {!ev.locationLink && ev.location}
              </div>
              <h3
                className={cx({
                  'fx aiCtr ttCap': true,
                  mB1: props.isEditable,
                  fs4: props.horizontal,
                })}>
                <Link
                  className="noUnderline"
                  to={ev.isDraft
                    ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                    : `/event/${ev.id}`}>
                  {ev.title}
                </Link>
              </h3>
              {props.showRSVPs && (
                <div
                  className={cx({
                    'fx aiCtr fs6 lh1': true,
                    hide: props.horizontal,
                    o5: isPastEvent,
                  })}>
                  {!ev.isDraft && <RSVP event={ev} />}
                  {ev.isDraft && 'This event is not published yet. Click to edit.'}
                </div>
              )}
              {props.showOrgName
                && (
                  <Link
                    to={`/org/${ev.orgId}`}
                    className={cx({
                      'fw600 fs6 lh1': true,
                      o5: isPastEvent,
                    })}>
                    {ev.orgName}
                  </Link>
                )}
              {!isPastEvent
              && props.isEditable
              && (
                <div className="fx aiCtr mB1">
                  <div className="bgGrey4 br4 fs7 lh1 mR2 p1 pL2 pR2 white">
                    {ev.isDraft ? 'Draft' : 'Published'}
                  </div>
                  <div className="fx aiCtr mR1">
                    <Link
                      to={`/org/${ev.orgId}/planMeeting?${objToQueryString(ev)}`}
                      className="btn fs7 fw600 hvrBgGrey1 lh1 noUnderline p1 pL2 pR2">
                      <span
                        role="img"
                        aria-label="Hand with Pen Emoji">
                        ✍️
                      </span>
                      Edit
                    </Link>
                  </div>
                  <button
                    onClick={e => props.deleteEvent(e, ev.id)}
                    className="aiCtr fs7 fw600 fx hvrBgGrey1 lh1">
                    <span
                      role="img"
                      className="mR1"
                      aria-label="Big X Emoji">
                      ✖️
                    </span>
                    Delete
                  </button>
                </div>
              )}
            </div>
          </div>
        </li>
      );
    })}
  </ul>
));
