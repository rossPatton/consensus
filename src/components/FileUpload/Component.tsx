import cx from 'classnames';
import React, {memo} from 'react';
import {v1} from 'uuid';

import {Emoji} from '~app/components';

import {tComponentProps} from './_types';

const baseUrl = 'https://consensus.nyc3.digitaloceanspaces.com';

export const FileUploadComponent = memo((props: tComponentProps) => (
  <>
    <h3 className="text-base mb-1">
      {props.title}
    </h3>
    <div
      id="imagePreview"
      className={cx({
        'animated bg-gray-1 p-3 mb-3 flex flex-col items-center': true,
        'fadeInDown': props.img,
        'hidden': !props.img,
      })}>
      <button
        type="button"
        onClick={props.removeImage}
        className="mb-2 p-2">
        <Emoji
          label="Close Emoji"
          emoji="✖️"
        />
        Remove Image
      </button>
      {props.img && (
        <img
          alt="Your uploaded file preview."
          height={props.width}
          width={props.width}
          src={`${baseUrl}/${props.folder}/${props.img}${__DEV__ ? `?v=${v1()}` : ''}`}
          className={cx({
            'rounded w-full': !props.fieldKey.includes('Avatar'),
            'rounded-circ': props.fieldKey.includes('Avatar'),
          })}
        />
      )}
    </div>
    <label
      htmlFor="fileUpload"
      className={cx({
        'fadeInDown bg-gray-1 border cursor-ptr flex flex-col items-center mb-3 p-3 rounded animated': true,
        'fadeInDown': !props.img,
        'hidden': props.img,
      })}>
      <input
        accept="image/png, image/jpeg, image/gif"
        id="fileUpload"
        name="eventFeaturedImage"
        onChange={props.setImage}
        type="file"
        className={cx({
          hidden: true,
        })}
      />
      <span className="btn flex items-center mb-1 p-1">
        <Emoji
          label="Picture (Camera) Emoji"
          emoji="📷"
        />
        Upload Your Image
      </span>
      {props.info && (
        <small>{props.info}</small>
      )}
    </label>
  </>
));
