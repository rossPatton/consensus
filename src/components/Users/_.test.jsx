import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import render from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import {testGroup1, testUser1} from '~app/constants/jest';

import Users from '.';
import {UsersComponent} from './Component';

const mockStore = configureStore([thunk]);

describe('components/Users', () => {
  it('renders container', () => {
    const store = mockStore({
      isLoading: false,
      usersByGroupId: {data: []},
    });

    const component = render.create((
      <Provider store={store}>
        <Users group={testGroup1} />
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders container alternative props', () => {
    const store = mockStore({
      isLoading: false,
      usersByGroupId: {
        data: [{
          id: 1,
          role: 'member',
        }, {
          id: 2,
          role: 'pending',
        }],
      },
    });

    const component = render.create((
      <Provider store={store}>
        <Users
          group={testGroup1}
          type="pending"
        />
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty component', () => {
    const component = render.create((
      <MemoryRouter>
        <UsersComponent
          group={testGroup1}
          isEditable
          isDesktop
          isMobile={false}
          items={[]}
          onRoleFilterChange={jest.fn()}
          onSearchChange={jest.fn()}
          removeUser={jest.fn()}
          showMobileControls={1}
          toggleMobileControls={jest.fn()}
          type="pending"
          users={[testUser1, testUser1]}
        />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty component alternative props', () => {
    const component = render.create((
      <MemoryRouter>
        <UsersComponent
          group={testGroup1}
          isEditable
          isDesktop={false}
          isMobile
          items={[]}
          onRoleFilterChange={jest.fn()}
          onSearchChange={jest.fn()}
          removeUser={jest.fn()}
          showMobileControls={1}
          toggleMobileControls={jest.fn()}
          type="members"
          users={[testUser1, testUser1]}
        />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders component with users', () => {
    const component = render.create((
      <MemoryRouter>
        <UsersComponent
          group={testGroup1}
          isEditable
          isDesktop={false}
          isMobile
          items={[testUser1, testUser1, testUser1]}
          onRoleFilterChange={jest.fn()}
          onSearchChange={jest.fn()}
          removeUser={jest.fn()}
          showMobileControls={1}
          toggleMobileControls={jest.fn()}
          type="members"
          users={[testUser1, testUser1]}
        />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
