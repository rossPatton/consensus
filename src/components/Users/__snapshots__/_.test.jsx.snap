// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Users renders component 1`] = `
<ul>
  <li
    className="bg-white hover:bg-gray-2 rounded flex items-center mb-1 p-1 relative"
  >
    <div
      className="flex d:absolute r d:mr-2 p-1 d:p-0"
    >
      <button
        className="border-0 bg-0 mr-2"
        onClick={[Function]}
      >
        Remove
      </button>
      <select
        onChange={[Function]}
      >
        <option>
          Change role
        </option>
        <option
          value="member"
        >
          member
        </option>
        <option
          value="facilitator"
        >
          facilitator
        </option>
      </select>
    </div>
    <div
      className="flex items-center"
    >
      <div>
        <div
          className="text-sm flex items-center text-sm leading-none"
        >
          <b
            className="mr-2"
          >
            facilitator
          </b>
        </div>
        <h3>
          <a
            href="/user/1"
            onClick={[Function]}
          >
            @
          </a>
        </h3>
      </div>
    </div>
  </li>
  <li
    className="bg-white hover:bg-gray-2 rounded flex items-center mb-1 p-1 relative"
  >
    <div
      className="flex d:absolute r d:mr-2 p-1 d:p-0"
    >
      <button
        className="border-0 bg-0 mr-2"
        onClick={[Function]}
      >
        Remove
      </button>
      <select
        onChange={[Function]}
      >
        <option>
          Change role
        </option>
        <option
          value="member"
        >
          member
        </option>
        <option
          value="facilitator"
        >
          facilitator
        </option>
      </select>
    </div>
    <div
      className="flex items-center"
    >
      <div>
        <div
          className="text-sm flex items-center text-sm leading-none"
        >
          <b
            className="mr-2"
          >
            facilitator
          </b>
        </div>
        <h3>
          <a
            href="/user/1"
            onClick={[Function]}
          >
            @
          </a>
        </h3>
      </div>
    </div>
  </li>
</ul>
`;

exports[`components/Users renders component alternative props 1`] = `
<ul>
  <li
    className="bg-white hover:bg-gray-2 rounded flex items-center mb-1 p-1 relative"
  >
    <button
      className="absolute r p-1 mr-1"
      onClick={[Function]}
    >
      Edit
    </button>
    <div
      className="flex items-center"
    >
      <div>
        <div
          className="text-sm flex items-center text-sm leading-none"
        >
          <b
            className="mr-2"
          >
            facilitator
          </b>
        </div>
        <h3>
          <a
            href="/user/1"
            onClick={[Function]}
          >
            @
          </a>
        </h3>
      </div>
    </div>
  </li>
  <li
    className="bg-white hover:bg-gray-2 rounded flex items-center mb-1 p-1 relative"
  >
    <button
      className="absolute r p-1 mr-1"
      onClick={[Function]}
    >
      Show
    </button>
    <div
      className="flex d:absolute r d:mr-2 p-1 d:p-0"
    >
      <button
        className="border-0 bg-0 mr-2"
        onClick={[Function]}
      >
        Remove
      </button>
      <select
        onChange={[Function]}
      >
        <option>
          Change role
        </option>
        <option
          value="member"
        >
          member
        </option>
        <option
          value="facilitator"
        >
          facilitator
        </option>
      </select>
    </div>
    <div
      className="flex items-center hidden"
    >
      <div>
        <div
          className="text-sm flex items-center text-sm leading-none"
        >
          <b
            className="mr-2"
          >
            facilitator
          </b>
        </div>
        <h3>
          <a
            href="/user/1"
            onClick={[Function]}
          >
            @
          </a>
        </h3>
      </div>
    </div>
  </li>
</ul>
`;

exports[`components/Users renders container 1`] = `"No users found that meet this criteria."`;

exports[`components/Users renders container alternative props 1`] = `null`;
